src/input_cp2k_global.F

```diff
@@ -99,28 +106,41 @@ SUBROUTINE create_global_section(section)
 
       CALL keyword_create(keyword, __LOCATION__, name="BLACS_REPEATABLE", &
                           description="Use a topology for BLACS collectives that is guaranteed to be repeatable "// &
-                          "on homegeneous architectures", &
+                          "on homogeneous architectures", &
                           usage="BLACS_REPEATABLE", &
                           default_l_val=.FALSE., lone_keyword_l_val=.TRUE.)
       CALL section_add_keyword(section, keyword)
       CALL keyword_release(keyword)
 
+#if defined(__ELPA)
+      default_i_val = do_diag_elpa
+      ! TODO (issue 391): parse e.g. CP2K_PREFERRED_DIAG_LIBRARY="SL" or CP2K_PREFERRED_DIAG_LIBRARY="ELPA"
+      CALL GET_ENVIRONMENT_VARIABLE("CP2K_DEFAULT_DIAG", env, STATUS=env_i)
```

Comment:
As discussed before, I'm very skeptical of using environment variables to overwrite the input.

Reply:
The isuee is that ELPA sometimes fails in decomposing the problem. I think you are right with this particular environment variable. It sort of a quick toggle to check whether an eventual issue is caused by ELPA. On the other hand one could just set `P2K_PREFERRED_DIAG_LIBRARY SL` in the input. I will remove this snippet, but I still think a general solution for runtime settings like mapping GLOBALs to environment variables is desirable.

Reply:
I think a hierarchy of the type (from strong to weak) for variables in GLOBAL is possible

1) Variable set in input
2) Variable set in environment
3) default set in code

Reply:
Personally I would use the following order since it's what most open source tools do in my experience:

1. Command line argument (if any, I think it would be very useful)
2. Variable set in environment
3. Variable set in input (configuration file)
4. Default by code

Reply:
I agree with Jürg. The Input files should have the highest priority, because it is not a configuration file, it's our main UI.

I think, instead of environments variables we should do more PR's like this which uses the hierarchy:
1. Variable set in input
2. Default overwritten by ARCH file flags
3. Default by code